Jenkinsfile

```diff
@@ -14,21 +14,78 @@ pipeline {
   }
   stages {
 
+    stage('Check for Important Changes') {
+      when {
+        not {
+          anyOf {
+            changeset "docker/**"
+            changeset "docs/**"
+            changeset "pom.xml"
+            changeset "src/main/**"
+            triggeredBy cause: 'UserIdCause'
+          }
+        }
+      }
+      steps {
+        echo 'No relevant changes detected. Skipping the rest of the pipeline.'
+        script {
+          currentBuild.result = 'NOT_BUILT'
+          error("No changes in important paths. Pipeline aborted.")
+        }
+      }
+    }
+
+    stage('Trivy Code Scan (Dependencies)') {
+      steps {
+        script {
+          sh '''
+            trivy fs --scanners vuln,secret,misconfig --output trivy-fs-report.txt .
+          '''
+        }
+      }
+    }
+
     stage('Building images') {
       steps{
         script {
           echo 'Pulled - ' + env.GIT_BRANCH
           devImage = docker.build( devRegistry, "-f ./docker/dev.dockerfile .")
           deplImage = docker.build( deplRegistry, "-f ./docker/depl.dockerfile .")
-          testImage = docker.build( testRegistry, "-f ./docker/test.dockerfile .")
+         // testImage = docker.build( testRegistry, "-f ./docker/test.dockerfile .")
+        }
+      }
+    }
+
+    stage('Trivy Docker Image Scan and Report') {
+      steps {
+        script {
+          sh "trivy image --output trivy-dev-image-report.txt ${devImage.imageName()}"
+          sh "trivy image --output trivy-depl-image-report.txt ${deplImage.imageName()}"
+        }
+      }
+      post {
+        always {
+          archiveArtifacts artifacts: 'trivy-*.txt', allowEmptyArchive: true
+          publishHTML(target: [
+            allowMissing: true,
+            keepAll: true,
+            reportDir: '.',
+            reportFiles: 'trivy-fs-report.txt, trivy-dev-image-report.txt, trivy-depl-image-report.txt',
+            reportName: 'Trivy Reports'
+          ])
         }
       }
     }
 
+
     stage('Unit Tests and Code Coverage Test'){
       steps{
         script{
-          sh 'docker compose -f docker-compose.test.yml up test'
+          //sh 'docker compose -f docker-compose.test.yml up test'
```

Comment:
remove commented lines and also cleanup the docker compose, the test service is not needed anymore.